docs/website/docs/dlt-ecosystem/verified-sources/rest_api.md

```diff
@@ -0,0 +1,527 @@
+
+### Resource configuration
+
+A resource configuration has the following fields:
+
+- `endpoint`: The endpoint configuration for the resource. It can be a string or a dict representing the endpoint settings. See the [endpoint configuration](#endpoint-configuration) section for more details.
+- `write_disposition`: The write disposition for the resource.
+- `primary_key`: The primary key for the resource.
+- `include_from_parent`: A list of fields from the parent resource to be included in the resource output.
+- `selected`: A flag to indicate if the resource is selected for loading. This could be useful when you want to load data only from child resources and not from the parent resource.
+
+### Endpoint configuration
+
+The endpoint configuration defines how to query the API endpoint. Quick example:
+
+```py
+{
+    "path": "issues",
+    "method": "GET",
+    "params": {
+        "sort": "updated",
+        "direction": "desc",
+        "state": "open",
+        "since": {
+            "type": "incremental",
+            "cursor_path": "updated_at",
+            "initial_value": "2024-01-25T11:21:28Z",
+        },
+    },
+    "data_selector": "results",
+}
+```
+
+The fields in the endpoint configuration are:
+
+- `path`: The path to the API endpoint.
+- `method`: The HTTP method to be used. Default is `GET`.
+- `params`: Query parameters to be sent with each request. For example, `sort` to order the results.
+- `json`: The JSON payload to be sent with the request (for POST and PUT requests).
+- `paginator`: Pagination configuration for the endpoint. See the [pagination](#pagination) section for more details.
+- `data_selector`: A JSONPath to select the data from the response. See the [data selection](#data-selection) section for more details.
+- `response_actions`: A list of actions that define how to process the response data.
+- `incremental`: Configuration for incremental loading.
+
+### Pagination
+
+The REST API source will try to automatically handle pagination for you. This works by detecting the pagination details from the first API response.
+
+In some special cases, you may need to specify the pagination configuration explicitly.
+
+These are the available paginator types:
+
+| Paginator class | String Alias (`type`) | Description |
+| -------------- | ------------ | ----------- |
+| [JSONResponsePaginator](../../general-usage/http/rest-client.md#jsonresponsepaginator) | `json_response` | The links to the next page are in the body (JSON) of the response. |
+| [HeaderLinkPaginator](../../general-usage/http/rest-client.md#headerlinkpaginator) | `header_link` | The links to the next page are in the response headers. |
+| [OffsetPaginator](../../general-usage/http/rest-client.md#offsetpaginator) | `offset` | The pagination is based on an offset parameter. With total items count either in the response body or explicitly provided. |
```

Comment:
offset will not instantiate from a shorthand string (I have a special exception with an explanation) but now I defined the typed dicts for each paginator so you can refer to them

Reply:
correct, but we're not mentioning this in the docs. maybe we need to consider if we want to keep the shortcut